src/main/java/com/gotocompany/depot/http/request/util/JsonParserUtils.java

```diff
@@ -0,0 +1,115 @@
+            case NULL:
+                return parseInternal(jsonNode, parsedMessage);
+            default:
+                throw new IllegalArgumentException("The provided Json type is not supported");
+        }
+    }
+
+    public static JsonNode parseInternal(ObjectNode objectNode, ParsedMessage parsedMessage) {
+        ObjectNode finalJsonObject = new JsonNodeFactory(false).objectNode();
+        for (Map.Entry<String, JsonNode> entry : objectNode.properties()) {
+            String keyString = entry.getKey();
+            TextNode keyStringNode = new TextNode(keyString);
+            JsonNode parsedKeyNode = parseInternal(keyStringNode, parsedMessage);
+            String parsedKeyString = parsedKeyNode.toString();
+            if (parsedKeyNode.getNodeType().equals(JsonNodeType.STRING)) {
+                parsedKeyString = parsedKeyString.substring(1, parsedKeyString.length() - 1);
+            }
+            JsonNode valueNode = entry.getValue();
+            JsonNode parsedValue = JsonParserUtils.parse(valueNode, parsedMessage);
+            finalJsonObject.put(parsedKeyString, parsedValue);
+        }
+        return finalJsonObject;
+    }
+
+
+    public static JsonNode parseInternal(ArrayNode arrayNode, ParsedMessage parsedMessage) {
+        ArrayNode tempJsonArray = new JsonNodeFactory(false).arrayNode();
+        for (Iterator<JsonNode> it = arrayNode.elements(); it.hasNext();) {
+            JsonNode jsonElement1 = it.next();
+            JsonNode parsedJsonNode = JsonParserUtils.parse(jsonElement1, parsedMessage);
+            tempJsonArray.add(parsedJsonNode);
+        }
+        return tempJsonArray;
+    }
+
+    public static JsonNode parseInternal(TextNode textNode, ParsedMessage parsedMessage) {
+        Template templateValue;
+        try {
+            templateValue = new Template(textNode.asText());
+        } catch (InvalidTemplateException e) {
+            throw new IllegalArgumentException(e.getMessage());
+        }
+        Object parsedValue = templateValue.parseWithType(parsedMessage);
+
+        String parsedJsonString = parsedValue.toString();
+
+        JsonNode parsedJsonNode;
+        if (parsedValue instanceof String) {
+            if (parsedJsonString.startsWith("\"") && parsedJsonString.endsWith("\"")) {
+                parsedJsonString = parsedJsonString.substring(1, parsedJsonString.length() - 1);
+            }
+            parsedJsonNode = new JsonNodeFactory(false).textNode(parsedJsonString);
+            return parsedJsonNode;
+        }
+        try {
+            parsedJsonNode = OBJECT_MAPPER.readTree(parsedJsonString);
+        } catch (JsonProcessingException e) {
+            throw new ConfigurationException("An error occurred while parsing the template string : " + parsedJsonString + "\nError: " + e.getMessage());
+        }
+        return parsedJsonNode;
+    }
+
+    public static JsonNode parseInternal(JsonNode jsonElement, ParsedMessage parsedMessage) {
```

Comment:
Lets not introduce too many terms; i.e. 

JsonNode jsonNode;      // correct
JsonNode jsonElement; // WRONG

Reply:
fixed